### PR TITLE
CSR updates for priv_1_12

### PR DESCRIPTION
**Pull request by silabs-hfegran**
Signed-off-by: Henrik Fegran <Henrik.Fegran@silabs.com>

### CODE REVIEW

File: cv32e40x/env/corev-dv/cv32e40x_csr_template.yaml
Comment (Silabs-ArjanB) [inline cv32e40x/env/corev-dv/cv32e40x_csr_template.yaml:939]: tcontrol does not belong in this category. It is a MRW CSR. (The same holds for tinfo; both of them are misdocumented in the UM; I will add an issue for that)

**Review by Silabs-ArjanB — CHANGES_REQUESTED** [review]

File: cv32e40x/env/corev-dv/cv32e40x_csr_template.yaml
Comment (silabs-hfegran) [reply cv32e40x/env/corev-dv/cv32e40x_csr_template.yaml:939]: Thanks and well spotted!
I will update this accordingly

**Comment by silabs-hfegran** [comment]
This is ready for review 

There are however two requirements for this commit to be merged: 
* https://github.com/google/riscv-dv/pull/843 must be merged to upstream riscv-dv first
* https://github.com/openhwgroup/cv32e40x/pull/396 must be merged to cv32e40x:master

Lacking updated register definitions in riscv-dv will lead to breakage of ___all___ corev-dv tests - so please do not merge until that is in place. Before merge takes place, this PR should be updated with the hash of `riscv-dv` and `cv32e40x:master` that corresponds to having the aforementioned fixes implemented.

With these fixes in place, this passes ci_check, and also resolves the `corev_rand_debug_ebreak` failures observed in regressions.

File: cv32e40x/env/corev-dv/cv32e40x_csr_template.yaml
Comment (silabs-robin) [inline cv32e40x/env/corev-dv/cv32e40x_csr_template.yaml:369]: Is all of this commented-out code autogenerated?
Where does it come from?

File: cv32e40x/env/uvme/uvme_cv32e40x_cfg.sv
Comment (silabs-robin) [inline cv32e40x/env/uvme/uvme_cv32e40x_cfg.sv:132]: Do we want to call it "master" version? The ISS does it, but we could (if we want to) call it something like v1.12-beta01 etc? Or do we want "master" to generally map to whatever the ISS "master" means, always?

File: cv32e40x/env/corev-dv/cv32e40x_privil_reg.sv
Comment (silabs-robin) [inline cv32e40x/env/corev-dv/cv32e40x_privil_reg.sv:42]: Are these fields removed because they are irrelevant to the 40x?

File: lib/uvm_agents/uvma_core_cntrl/uvma_core_cntrl_cfg.sv
Comment (silabs-robin) [inline lib/uvm_agents/uvma_core_cntrl/uvma_core_cntrl_cfg.sv:184]: Is this chosen as the default to maintain backwards compatibility?

**Comment by silabs-robin** [comment]
I don't have enough knowledge of the new priv spec or of riscv-dv to have a qualified opinion on the correctness of everything in this PR.
But I have read through it all and did not see anything that I think looks odd, apart from my questions above.

File: cv32e40x/env/corev-dv/cv32e40x_csr_template.yaml
Comment (silabs-hfegran) [reply cv32e40x/env/corev-dv/cv32e40x_csr_template.yaml:369]: It is _not_ autogenerated, I wrote this manually. The reason why I kept the non-existent registers present, albeit commented out,  was to ease integration with other cores that could potentially use these.

File: cv32e40x/env/uvme/uvme_cv32e40x_cfg.sv
Comment (silabs-hfegran) [reply cv32e40x/env/uvme/uvme_cv32e40x_cfg.sv:132]: What ISS calls master points to a certain revision of the spec on which it is based. If we changed the name of this, then it could easily get misrepresented if we update the ISS. That is the reasoning for keeping the original naming.

File: cv32e40x/env/corev-dv/cv32e40x_privil_reg.sv
Comment (silabs-hfegran) [reply cv32e40x/env/corev-dv/cv32e40x_privil_reg.sv:42]: They are removed because they are deprecated from the spec - these field definitions no longer exist (changed to WPRI in spec)

File: lib/uvm_agents/uvma_core_cntrl/uvma_core_cntrl_cfg.sv
Comment (silabs-hfegran) [reply lib/uvm_agents/uvma_core_cntrl/uvma_core_cntrl_cfg.sv:184]: Yes, this is chosen to present the exact same version of the privilege spec as before, to keep compatibility with e.g. cv32e40p without having to update code specific to that core. Cores that require another version must set this constraint in its environment

**Comment by silabs-robin** [comment]
The ExternalRepos.mk change looks good to me (even if it is a tiny bit dissatisfying that we must temporarily point to a non-official fork).
Based also on my previous read-through I want to approve.
But the CV_CORE_HASH needs to be updated after the RTL PR is merged, before this PR can be merged?

**Comment by silabs-hfegran** [comment]
> But the CV_CORE_HASH needs to be updated after the RTL PR is merged, before this PR can be merged?

Agreed - core code needs to be merged first, followed by hash update then this PR can be merged

**Review by Silabs-ArjanB — COMMENTED** [review]
I added some questions

File: cv32e40x/env/corev-dv/target/cv32e40x/riscv_core_setting.sv
Comment (silabs-hfegran) [reply cv32e40x/env/corev-dv/target/cv32e40x/riscv_core_setting.sv:104]: These have to be added commented out (or updated in riscv-dv) - as adding CSRs here that do not exist in riscv-dv will cause compilation failure.

File: cv32e40x/env/corev-dv/cv32e40x_privil_reg.sv
Comment (silabs-hfegran) [reply cv32e40x/env/corev-dv/cv32e40x_privil_reg.sv:49]: This should be a relatively easy thing to implement, and all required changes are local to corev-dv. To my knowledge, the only use for these bitfield definitions is to set up the initial value for mstatus during initialization, the fields are otherwise unused. 

Similarly, the presence (or lack thereof) of the UIE field here only affects the initial value of mstatus, and it is by default set to 0 which is still the case when this is treated as a non-existing field. (see code below, `setup_mmode_reg()` is the only applicable code that uses this register definition). 

I suggest we add suitable defaults for the VS, FS, XS and SD bits when they are added, there is no need to add them here until they actually need a default reset value.

**Review by Silabs-ArjanB — APPROVED** [review]

**Pull request merged**
